var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -91,6 +93,19 @@ def edit(self, spec, prefix):
         os.environ['ESMF_INSTALL_LIBDIR'] = 'lib'
         os.environ['ESMF_INSTALL_MODDIR'] = 'include'
 
+        # Allow compiler flags to carry through from compiler spec
+        os.environ['ESMF_CXXCOMPILEOPTS'] = \
+            ' '.join(spec.compiler_flags['cxxflags'])
+        os.environ['ESMF_F90COMPILEOPTS'] = \
+            ' '.join(spec.compiler_flags['fflags'])
+        # ESMF will simply not build with Intel using backing GCC 8, in that
+        # case you need to point to something older, below is commented but is
+        # an example
```

Comment:
We need a better solution to this than comments in the `package.py`.  @alalazo  do you have any ideas on how to get the right behavior using compiler specs?  It looks like ESMF works with `intel-with-gcc7` (for example), but not `intel-with-gcc8`.  The core problem is the Intel compiler is not a single compiler; but rather, it leverages off of a GCC back end.  So the same installation of Intel is essentially a different compiler, depending on which GCC you pair it with.